x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/expression/function/fulltext/MatchFunction.java

```diff
@@ -0,0 +1,132 @@
+import org.elasticsearch.xpack.esql.core.expression.TypeResolutions;
+import org.elasticsearch.xpack.esql.core.querydsl.query.MatchQuery;
+import org.elasticsearch.xpack.esql.core.querydsl.query.Query;
+import org.elasticsearch.xpack.esql.core.querydsl.query.QueryStringQuery;
+import org.elasticsearch.xpack.esql.core.tree.NodeInfo;
+import org.elasticsearch.xpack.esql.core.tree.Source;
+import org.elasticsearch.xpack.esql.expression.function.Example;
+import org.elasticsearch.xpack.esql.expression.function.FunctionInfo;
+import org.elasticsearch.xpack.esql.expression.function.Param;
+import org.elasticsearch.xpack.esql.io.stream.PlanStreamInput;
+
+import java.io.IOException;
+import java.util.List;
+
+import static org.elasticsearch.xpack.esql.core.expression.TypeResolutions.ParamOrdinal.FIRST;
+import static org.elasticsearch.xpack.esql.core.expression.TypeResolutions.ParamOrdinal.SECOND;
+import static org.elasticsearch.xpack.esql.core.expression.TypeResolutions.isNotNull;
+import static org.elasticsearch.xpack.esql.core.expression.TypeResolutions.isString;
+
+/**
+ * Full text function that performs a {@link QueryStringQuery} .
+ */
+public class MatchFunction extends FullTextFunction implements Validatable {
+
+    public static final NamedWriteableRegistry.Entry ENTRY = new NamedWriteableRegistry.Entry(
+        Expression.class,
+        "Match",
+        MatchFunction::new
+    );
+
+    private final Expression field;
+
+    @FunctionInfo(
+        returnType = "boolean",
+        preview = true,
+        description = "Performs a match query on the specified field. Returns true if the provided query matches the row.",
+        examples = { @Example(file = "match-function", tag = "match-with-field") }
+    )
+    public MatchFunction(
+        Source source,
+        @Param(name = "field", type = { "keyword", "text" }, description = "Field that the query will target.") Expression field,
+        @Param(
+            name = "query",
+            type = { "keyword", "text" },
+            description = "Text you wish to find in the provided field."
+        ) Expression matchQuery
+    ) {
+        super(source, matchQuery, List.of(matchQuery, field));
+        this.field = field;
+    }
+
+    private MatchFunction(StreamInput in) throws IOException {
+        this(Source.readFrom((PlanStreamInput) in), in.readNamedWriteable(Expression.class), in.readNamedWriteable(Expression.class));
+    }
+
+    @Override
+    public void writeTo(StreamOutput out) throws IOException {
+        source().writeTo(out);
+        out.writeNamedWriteable(field);
+        out.writeNamedWriteable(query());
+    }
+
+    @Override
+    public String getWriteableName() {
+        return ENTRY.name;
+    }
+
+    @Override
+    public String functionName() {
+        return "MATCH";
+    }
+
+    @Override
+    public Query asQuery(String queryText) {
+        return new MatchQuery(source(), ((FieldAttribute) field).name(), queryText);
```

Comment:
Move the build of `MatchQuery` into `EsqlExpressionTranslators`, it is a central place to build pushdown queries.

Reply:
That allows to simplify the methods in the superclass, done in 06f2eafd9177a8d2d8c80300072039ac9e79407e